tf2_sensor_msgs/test/test_tf2_sensor_msgs.py

```diff
@@ -29,61 +29,185 @@
+
         # checking no modification in input cloud
-        assert(old_data == self.point_cloud_in.data)
+        self.assertEqual(old_data, self.point_cloud_in.data)
+
+    def test_simple_rotation_transform(self):
+        # Create a transform containing only a 180° rotation around the x-axis
+        transform_rot = TransformStamped()
+        transform_rot.transform.rotation.x = 1.0
+        transform_rot.transform.rotation.w = 0.0
+
+        # Apply transform
+        point_cloud_transformed = do_transform_cloud(self.point_cloud_in, transform_rot)
+
+        # Expected output
+        expected_coordinates = np.array([[1, -2, 0], [10, -20, -30]], dtype=np.float32)
+
+        # Compare to ground truth
+        new_points = read_points_numpy(point_cloud_transformed)
+        self.assertTrue(np.allclose(expected_coordinates, new_points))
+
+    def test_rotation_and_translation_transform(self):
+        # Create a transform combining a 100m z translation with
+        # a 180° rotation around the x-axis
+        transform = TransformStamped()
+        transform.transform.translation.z = 100.0
+        transform.transform.rotation.x = 1.0
+        transform.transform.rotation.w = 6.123234e-17
+
+        # Apply transform
+        point_cloud_transformed = do_transform_cloud(
+            self.point_cloud_in, transform)
+
+        # Expected output
+        expected_coordinates = np.array([[1, -2, 100], [10, -20, 70]], dtype=np.float32)
+
+        # Compare to ground truth
+        new_points = read_points_numpy(point_cloud_transformed)
+        self.assertTrue(np.allclose(expected_coordinates, new_points))
+
+    def test_direct_transform(self):
+        # Create a transform combining a 100m z translation with
+        # a 180° rotation around the x-axis
+        transform = Transform()
+        transform.translation.z = 100.0
+        transform.rotation.x = 1.0
+        transform.rotation.w = 6.123234e-17
+
+        # Transform points
+        points = transform_points(self.points, transform)
+
+        # Expected output
+        expected_coordinates = np.array([[1, -2, 100], [10, -20, 70]], dtype=np.float32)
+
+        # Compare to ground truth
+        self.assertTrue(np.allclose(expected_coordinates, points))
+
+    def test_tf2_ros_transform(self):
+        # Our target frame
+        target_frame_name = "base_footprint"
+
+        # We need to create a local test tf buffer
+        tf_buffer = tf2_ros.Buffer()
+
+        # We need to fill this tf_buffer with a possible transform
+        # So we create a transform with a 100m z translation
+        transform = TransformStamped()
+        transform.header.frame_id = "test"
+        transform.child_frame_id = target_frame_name
+        transform.transform.translation.z = 100.0
+        transform.transform.rotation.w = 1.0
+
+        # Set the new transform in our local tf_buffer
+        tf_buffer.set_transform_static(transform, "unittest")
```

Comment:
Nit: single quotes preferred according to the style guide

```suggestion
        tf_buffer.set_transform_static(transform, 'unittest')
```